src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -376,6 +379,63 @@ public async Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestP
             }
         }
 
+        /// <summary>
+        ///     Parse out the owner and repo from a repository url
+        /// </summary>
+        /// <param name="prBody">Pull Request commit msg</param>
+        /// <returns>Tuple of owner and repo</returns>
+        public static string ParsePullRequestBody(string prBody)
+        {
+            var matches = DependencyUpdatesPattern.Matches(prBody);
+            if (matches.Count == 0)
+            {
+                return prBody;
+            }
+            string message = "";
+            foreach (Match match in matches)
+            {
+                message += $@"
+{match.Value.ToString().Replace("*", "")}";
+            }
+            return message;
+        }
+
+        /// <summary>
+        ///     Merge a dependency update pull request
+        /// </summary>
+        /// <param name="pullRequestUrl">Uri of pull request to merge</param>
+        /// <param name="parameters">Settings for merge</param>
+        /// <returns></returns>
+        public async Task MergeDependencyPullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters)
+        {
+            (string owner, string repo, int id) = ParsePullRequestUri(pullRequestUrl);
+            Octokit.PullRequest pr = await Client.PullRequest.Get(owner, repo, id);
+            var commits = await Client.PullRequest.Commits(owner, repo, pr.Number);
+            string commitMessage = $@"{pr.Title}
+{ParsePullRequestBody(pr.Body)}";
+            foreach (Octokit.PullRequestCommit commit in commits)
+            {
+                if (!commit.Commit.Author.Name.Equals("dotnet-maestro[bot]"))
+                {
```

Comment:
Commit message is basically the PR description but if there are any manual commits ( I can identify that by checking the author of the commit if its not "dotnet-maestro[bot]") then I append the commit msg (i.e, PR description) with the manual commit message.

Reply:
What do you think about this?